style_guides/js_style_guide.md

```diff
@@ -153,188 +185,288 @@ Variables, properties and function names should use `lowerCamelCase`.  They
+should never be imported directly. Instead, you must only import the top-level
+API that's exported by the module itself.
+
+On the other hand, a module should be able to import parent and sibling
+modules.
 
 ```js
-if (width < 300) {
-  ...
-}
+// good
+import foo from 'foo';
+import child from './child';
+import parent from '../';
+import ancestor from '../../../';
+import sibling from '../foo';
+
+// bad
+import inFoo from 'foo/child';
+import inSibling from '../foo/child';
 ```
 
 ## Global definitions
 
-Don't do this. Everything should be wrapped in a module that can be depended on by other modules. Even things as simple as a single value should be a module.
+Don't do this. Everything should be wrapped in a module that can be depended on
+by other modules. Even things as simple as a single value should be a module.
 
 ## Function definitions
 
-Prefer the use of function declarations over function expressions. Function expressions are allowed, but should usually be avoided.
-
-Also, keep function definitions above other code instead of relying on function hoisting.
+Use function declarations over function expressions.
 
-*Preferred:*
+Also, keep function definitions above other code instead of relying on function
+hoisting.
 
 ```js
+// good
 function myFunc() {
   ...
 }
+
+// bad
+const myFunc = function () {
+  ...
+};
 ```
 
-*Allowed:*
+## Arrow functions
+
+If you must use a function expression, then use an arrow function:
 
 ```js
-var myFunc = function () {
-  ...
-};
+// good
+[1, 2, 3].map((n) => {
+  const m = doSomething(n);
+  return m - n;
+})
+
+// bad
+[1, 2, 3].map(function (n) {
+  const m = doSomething(n);
+  return m - n;
+})
+```
+
+If your function body does not include braces and only accepts one argument,
+then omit the argument paranthesis:
+
+```js
+// good
+[1, 2, 3].map(n => n + 1);
+
+// bad
+[1, 2, 3].map((n) => n + 1);
+
+// bad
+[1, 2, 3].map(n => {
+  return n + 1;
+});
+```
+
+If your arrow function is only returning an object literal, then wrap the
+object in paranthesis rather than using an explicit return:
+
+```js
+// good
+() => ({
+  foo: 'bar'
+})
+
+// bad
+() => {
+  return {
+    foo: 'bar'
+  };
+}
 ```
 
 ## Object / Array creation
 
 Use trailing commas and put *short* declarations on a single line. Only quote
 keys when your interpreter complains:
 
-*Right:*
-
 ```js
-var a = ['hello', 'world'];
-var b = {
+// good
+const a = ['hello', 'world'];
+const b = {
   good: 'code',
   'is generally': 'pretty'
 };
-```
 
-*Wrong:*
-
-```js
-var a = [
+// bad
+const a = [
   'hello', 'world'
 ];
-var b = {"good": 'code'
+const b = {'good': 'code'
         , is generally: 'pretty'
         };
 ```
 
 ## Object / Array iterations, transformations and operations
 
-Use native ES5 methods to iterate and transform arrays and objects where possible. Do not use `for` and `while` loops.
+Use native methods to iterate and transform arrays and objects where possible.
+Do not use `for` and `while` loops.
```

Comment:
Let's rephrase these two lines to be less draconian and to offer guidance on _why_:

```javascript
In many cases, you can use native array and object iteration and transformation methods
instead of `for` and `while` loops, to make your code clearer and terser.
```